### PR TITLE
Add PyGMT Zenodo BibTeX entry to main README.md

### LINKED ISSUE #653: How to cite PyGMT
**Description of the problem**

Searching for "PyGMT" in Google Scholar shows that there are already some papers citing/mentioning/acknowledging PyGMT. Here are the searching results: https://scholar.google.com/scholar?q=PyGMT&hl=en&as_sdt=0%2C23&as_ylo=2017&as_yhi=

From the searching results, you'll see that people are citing PyGMT in various ways:

1. citing the GMT 6 paper
2. citing @leouieda's AGU abstracts
3. citing the Zenodo stable DOI or DOI of a specific version.

PyGMT should document the correct/recommended way for citation.

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

To ensure that people will cite PyGMT consistently in their research work, this PR adds a PyGMT BibTeX entry to the main README.md (copied from Zenodo at https://zenodo.org/record/4025418/export/hx).

Also point to https://www.generic-mapping-tools.org/cite for GMT specific modules.

Fixes #653

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Notes**

- You can write `/format` in the first line of a comment to lint the code automatically

### CODE REVIEW

**Comment by seisman** [comment]
What about a stand-alone `CITATION.md` file in the repository, see https://github.com/GenericMappingTools/gmt/pull/115.

**Comment by weiji14** [comment]
> What about a stand-alone `CITATION.md` file in the repository, see [GenericMappingTools/gmt#115](https://github.com/GenericMappingTools/gmt/pull/115).

Don't think we need a dedicated page for this now since it's just one entry. We can point to https://pygmt-98df8j9um.vercel.app/#citing-pygmt (or https://pygmt.org/#citing-pygmt when this PR is merged) to instruct on how to cite PyGMT.

> Yes, I think we should encourage users to cite both PyGMT and GMT6 if possible.

:ok_hand:

**Comment by seisman** [comment]
The line breaks in the title and note are unnecessary:
image

**Comment by seisman** [comment]
> Don't think we need a dedicated page for this now since it's just one entry. We can point to https://pygmt-98df8j9um.vercel.app/#citing-pygmt (or https://pygmt.org/#citing-pygmt when this PR is merged) to instruct on how to cite PyGMT.

I'm feeling that the README.md (or the website landing page) is a little too long, and readers may miss the citation information.

**Comment by weiji14** [comment]
> > Don't think we need a dedicated page for this now since it's just one entry. We can point to https://pygmt-98df8j9um.vercel.app/#citing-pygmt (or https://pygmt.org/#citing-pygmt when this PR is merged) to instruct on how to cite PyGMT.
> 
> I'm feeling that the README.md (or the website landing page) is a little too long, and readers may miss the citation information.

I suppose it depends on what's better for the search engine (e.g. results from "How to cite PyGMT"). There's some tips at http://citeas.org/modify-your-citation. Currently looks like this at http://citeas.org/cite/pygmt:

image

It's a lot better for GMT at https://citeas.org/cite/gmt:

image

Personally I just do `CTRL+F` and type 'cite' or 'BibTeX' to find the citation info, but we'll need to optimize this for the machines :robot: too :smile:

File: README.rst [hunk 1]
```diff
@@ -145,13 +145,38 @@ mistakes. That's how we all improve and we are happy to help others learn.
...
+      month        = sep,
+      year         = 2020,
+      publisher    = {Zenodo},
+      version      = {v0.2.0},
+      doi          = {10.5281/zenodo.4025418},
+      url          = {https://doi.org/10.5281/zenodo.4025418}
+    }
+
+To cite a specific version of PyGMT, go to our Zenodo page at
+https://doi.org/10.5281/zenodo.3781524 and use the Export to BibTeX function there.
```
Comment (seisman) [inline README.rst:175]:
```suggestion
https://doi.org/10.5281/zenodo.3781524 and use the "Export to BibTeX" function there.
```

**Comment by weiji14** [comment]
Another argument **for** having the BibTeX in the main README.md is that it will appear on PyPI as well :wink:

**Review by seisman — APPROVED** [review]

**Comment by seisman** [comment]
Forgot to add "updating citation information" in the release checklist.

**Comment by weiji14** [comment]
Oh right, I'll merge in #673 first and we can do that, and probably start the changelog for PyGMT v0.2.1

**Pull request merged**